src-v2/NTS.Domain.Setup/Entities/Lap.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Newtonsoft.Json;
+
+namespace NTS.Domain.Setup.Entities;
+public class Lap : DomainEntity
+{
+    public static Lap Create(double distance) => new (distance);
+    public static Lap Update(int id, double distance) => new(id, distance);
+
+    [JsonConstructor]
+    public Lap(int id, double distance) : this(distance)
+    {
+        Id = id;
+    }
+    public Lap(double distance)
+    {
+        if (distance <= 0)
+        {
+            throw new DomainException(nameof(Distance), "Distance cannot be zero or less.");
+        }
+
+        Distance = distance;
+    }
+    public double Distance { get; set; }
+
+    public string NestedToString()
+    {
+        return Distance.ToString()+"km";
+    }
```

Comment:
NestedToString? Nested how? Is this ncessary?

Reply:
Nested because when Lap is a property of Phase the ToString() method of Phase would need to print only the Distance property of Lap. Alternatively, the Distance property can be called directly.

Reply:
That means that you are defining a member of Lap with the added context of Phase as it's parent. That's wrong. Lap doesn't know and shouldn't know how it is used. This is true for all objects, not only Domain ones.

ToString in Domain entities should be agnostic of their usage - i.e. they should provide relevant but not excessive information. In the example bellow "long" is completely unnecessary and the "Lap -" is questionable